.github/workflows/_build.yml

```diff
@@ -35,108 +32,63 @@ jobs:
       - name: Verify Bundler Version
         run: if [[ $(cat Gemfile.lock | sed -n '/BUNDLED WITH/{n;p;}' | sed '/^$/d;s/[[:blank:]]//g') != "2.2.3" ]]; then echo "Error - Gemfile.lock should be bundled with bundler 2.2.3"; exit 64; fi
 
-      - name: Set GIT_TAG
-        id: extract_git_tag
-        run: |
-          if [[ $GITHUB_REF == "refs/tags/"* ]]
-          then
-            echo ::set-output name=GIT_TAG::${GITHUB_REF#refs/tags/}
-          else
-            echo ::set-output name=GIT_TAG::""
-          fi
-
-      - name: Print tag
-        run: echo ${{ steps.extract_git_tag.outputs.GIT_TAG }}
-
-      - name: Select Xcode 13
-        run: sudo xcode-select --switch /Applications/Xcode_13.2.1.app
-
-      - name: nvm install
-        run: nvm install
-
       - name: Set Ruby version
         uses: ruby/setup-ruby@v1 # Sets ruby version via `.ruby-version`
 
-      - name: Set Bundler Config
-        run: bundle config path vendor/bundle
-
-      - name: Cache Bundler install Gems
-        uses: actions/cache@v3
-        with:
-          path: vendor/bundle
-          key: ${{ runner.os }}-gems-${{ env.ImageVersion }}-${{ hashFiles('**/Gemfile.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-gems-${{ env.ImageVersion }}
-
-      - name: Cache Pods
-        uses: actions/cache@v3
-        with:
-          path: Example/Pods
-          key: ${{ runner.os }}-pods-${{ hashFiles('**/Podfile.lock') }}
-          restore-keys: |
-            ${{ runner.os }}-pods-
+  Lint:
+    name: Lint Project
+    runs-on: macos-11
+    needs: [Setup]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
 
       - name: Cache Node modules
         uses: actions/cache@v3
+        id: cache_node_modules
         with:
           path: node_modules
           key: ${{ runner.os }}-node_modules-${{ hashFiles('**/package-lock.json') }}
           restore-keys: |
-            ${{ runner.os }}-node_modules-     
+            ${{ runner.os }}-node_modules-
 
-      - name: NPM ci
-        run: |
-          nvm use
-          npm ci
-
-      - name: Bundle Install
-        run: bundle install --jobs 4 --retry 3
-
-      - name: "Erase content from simulators"
-        run: "pkill Simulator || xcrun simctl erase all"
+      - name: nvm install
+        run: nvm install
 
-      - name: Pod Install
-        run: bundle exec pod install
-        working-directory: Example
+      - name: NPM ci
+        if: steps.cache_node_modules.outputs.cache-hit != 'true'
+        run: nvm use && npm ci
```

Comment:
Should this cache hit check be on the step below? What does npm ci do?

Reply:
`npm ci` install dependencies respecting package-lock.json, installing those exact versions.

This needs to run before running any npm command, so needs to be done here, as the very next step is running `npm run build`

the output of `npm ci` is the same as the cache_node_modules thing would cache